src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs

```diff
@@ -29,69 +26,26 @@ public X509CertificateCollection(X509CertificateCollection value)
             AddRange(value);
         }
 
-        public int Count
-        {
-            get { return _list.Count; }
-        }
-
-        bool ICollection.IsSynchronized
-        {
-            get { return false; }
-        }
-
-        object ICollection.SyncRoot
-        {
-            get { return NonGenericList.SyncRoot; }
-        }
-
-        bool IList.IsFixedSize
+        public X509Certificate this[int index] 
         {
-            get { return false; }
-        }
-
-        bool IList.IsReadOnly
-        {
-            get { return false; }
-        }
-
-        object IList.this[int index]
-        {
-            get
+            get 
             {
-                return NonGenericList[index];
+                return ((X509Certificate)(List[index]));
             }
-            set
+            set 
             {
                 if (value == null)
                     throw new ArgumentNullException(nameof(value));
 
-                NonGenericList[index] = value;
-            }
-        }
-
-        public X509Certificate this[int index]
-        {
-            get
-            {
-                return _list[index];
-            }
-            set
-            {
-                if (value == null)
-                    throw new ArgumentNullException(nameof(value));
+                OnInsert(index, value);
```

Comment:
It doesn't look like the desktop implementation calls `OnInsert`... What's the purpose of it here?